### ISSUE TITLE
🇧🇷 Função de imprimir não funciona / 🇺🇸 Printing doesn't work

### ISSUE THREAD

**Issue by guihkx** [flathub/br.gov.fazenda.receita.irpf2023#4]
## 🇧🇷 Como reproduzir o problema:

1. Tenha uma declaração pelo menos parcialmente preenchida
2. Aperte <kbd>Ctrl</kbd>+<kbd>P</kbd> para abrir a janela de impressão
3. Escolha qualquer declaração
4. Verifique que a opção `Visualizar` esteja selecionada
5. Aperte `Ok`
6. Aperte no botão com ícone de uma impressora, no canto superior esquerdo
7. O seguinte erro aparecerá:

    ![image](https://user-images.githubusercontent.com/626206/194819668-7fbb6ce0-6803-4056-82b7-0404cbc404e8.png)

Possível origem do bug: https://github.com/flatpak/flatpak/issues/3355

### Alternativas

1. Salve o documento como arquivo PDF em `~/Documents`
2. Abra o arquivo PDF usando um navegador web
3. Imprima o documento

---

## 🇺🇸 Steps to reproduce:

1. Have a tax form at least partially filed
2. Hit <kbd>Ctrl</kbd>+<kbd>P</kbd> to open the printing window
3. Select any tax form
4. Make sure the `Visualizar` radio button is the one selected
5. Hit `Ok`
6. Click the small printer button, located at the top left corner
7. Error pops up:

    ![image](https://user-images.githubusercontent.com/626206/194819668-7fbb6ce0-6803-4056-82b7-0404cbc404e8.png)
    TL;DR: *"Failed to communicate with the printer"*

Likely upstream bug: https://github.com/flatpak/flatpak/issues/3355

### Workarounds

1. Export the document as a PDF file to `~/Documents`
2. Open the PDF file with a web browser
3. Print it

**Comment by astroty** [comment]
Same problem here. Printing doesn't work.

**Comment by guihkx** [comment]
@astroty Have you tried adding the CUPS permission? I.e.:

```
flatpak override --user --socket=cups br.gov.fazenda.receita.irpf
```

Then restart the app and try printing again.

**Comment by astroty** [comment]
I've tried it through flatseal and gave it pretty much every permission and it didn't seem to wanna work
"Houve uma falha na comunicação com a impressora e não foi possível concluir a impressão.
Recomendamos salvar a declaração em PDF e imprimi-la novamente."
Or, if instead of clicking "Visualisar", I choose "Imprimir" instead, it gives me this different error message:
![image](https://user-images.githubusercontent.com/1356321/233404346-140ca50a-6bda-4a34-a4e6-edd79c5f1549.png)

The logs show
F: CUPS configuration file '/home/user/.cups/client.conf': Erro ao abrir arquivo /home/user/.cups/client.conf: Arquivo ou diretório inexistente
F: CUPS configuration file '/etc/cups/client.conf': Erro ao abrir arquivo /etc/cups/client.conf: Arquivo ou diretório inexistente

IRPF2023.log:
https://gist.github.com/astroty/669efab7a1cc52d63ef58ba2c3f058c7

**Comment by guihkx** [comment]
Yeah, that's unfortunate. The error messages you're getting seem to match the ones from the issue I linked in my first post. I'm not aware of a workaround yet.

**Comment by guihkx** [comment]
This will likely be fixed if the issue below gets addressed:

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1636

**Comment by Erick555** [comment]
It should be fixed in freedesktop 23.08.4+

**Comment by guihkx** [comment]
Confirming fixed. Thanks!
